Source/WebCore/platform/text/mac/TextBoundaries.mm

```diff
@@ -191,32 +233,47 @@ void findWordBoundary(StringView text, int position, int* start, int* end)
         *end = 0;
         return;
     }
-
-    unsigned pos = position;
-    if (pos >= text.length() && pos)
-        pos = text.length() - 1;
+    if (static_cast<unsigned>(position) >= text.length()) {
+        // This makes no sense. Return what we got in.
+        ASSERT(true);
+        *start = position;
+        *end = position;
+        return;
+    }
 
     // For complex text (Thai, Japanese, Chinese), visible_units will pass the text in as a 
     // single contiguous run of characters, providing as much context as is possible.
     // We only need one character to determine if the text is complex.
-    char32_t ch;
-    unsigned i = pos;
-    U16_NEXT(text, i, text.length(), ch);
-    bool isComplex = requiresContextForWordBoundary(ch);
+    char32_t character;
+    unsigned i = position;
+    bool is8Bit = text.is8Bit();
+    if (is8Bit)
+        character = text.span8()[position];
+    else
+        U16_NEXT(text, i, text.length(), character);
+    bool isComplex = requiresContextForWordBoundary(character);
 
     // FIXME: This check improves our word boundary behavior, but doesn't actually go far enough.
     // See <rdar://problem/8853951> Take complex word boundary finding path when necessary
     if (!isComplex) {
         // Check again for complex text, at the start of the run.
-        i = 0;
-        U16_NEXT(text, i, text.length(), ch);
-        isComplex = requiresContextForWordBoundary(ch);
+        if (is8Bit)
+            character = text.span8()[0];
+        else {
+            i = 0;
+            U16_NEXT(text, i, text.length(), character);
+        }
+        isComplex = requiresContextForWordBoundary(character);
     }
 
     if (isComplex)
         findComplexWordBoundary(text, position, start, end);
-    else
-        findSimpleWordBoundary(text, position, start, end);
+    else {
+        if (is8Bit)
+            findSimpleWordBoundary<true>(text, position, start, end);
+        else
+            findSimpleWordBoundary<false>(text, position, start, end);
+    }
```

Comment:
I think you can reuse the template arg
```suggestion
    else
        findSimpleWordBoundary<is8Bit>(text, position, start, end);
```

Reply:
It's not a template arg in this function though, so it has to be a literal. :)

Reply:
I thinkkkk this would still work, but again, just a nit anyway!

Reply:
I get it. I make the same mistake every time I have to use templates, and the compiler always tells me off. :p